scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+        ls_max_iter = min(max(n * self.MAXITER_RATIO, self.MAXITER_MIN),
+                          self.MAXITER_MAX)
+        self.kwargs['method'] = 'L-BFGS-B'
+        self.kwargs['options'] = {
+            'maxiter': ls_max_iter,
+        }
+        self.kwargs['bounds'] = list(zip(self.lower, self.upper))
+
+    def func_wrapper(self, x):
+        self.nb_fun_call += 1
+        return self.func(x, *self.fun_args)
+
+    def local_search(self, x):
+        x_tmp = np.copy(x)
+        fun_temp = self.func_wrapper(x)
+        mres = self.minimizer(self.func_wrapper, x, **self.kwargs)
+        # Check if is valid value
+        is_finite = np.all(np.isfinite(mres.x)) and np.isfinite(mres.fun)
+        in_bounds = np.all(mres.x >= self.lower) and np.all(
+            mres.x <= self.upper)
+        is_valid = is_finite and in_bounds
+
+        # Use the new point only if it is valid and return a better results
+        if is_valid and mres.fun < fun_temp:
+            return mres.fun, mres.x
+        else:
+            return fun_temp, x_tmp 
+
+
+def simulated_annealing(func, x0, bounds, maxiter=1000, local_search_options={},
+        initial_temp=5230., visit=2.62, accept=-5.0, maxfun=1e7, seed=None,
+        no_local_search=False):
+    """
+    Find the global minimum of a function using the Simulated Dual Annealing
+    algorithm
+
+    Parameters
+    ----------
+    func : callable
+        The objective function to be minimized.  Must be in the form
+        ``f(x, *args)``, where ``x`` is the argument in the form of a 1-D array
+        and ``args`` is a  tuple of any additional fixed parameters needed to
+        completely specify the function.
+    x0 : ndarray
+        The starting coordinates. If ``None`` is provided, initial
+        coordinates are automatically generated.
+    bounds : sequence
+        Bounds for variables.  ``(min, max)`` pairs for each element in ``x``,
+        defining the lower and upper bounds for the optimizing argument of
+        `func`. It is required to have ``len(bounds) == len(x)``.
+        ``len(bounds)`` is used to determine the number of parameters in ``x``.
+    maxiter : int, optional
+        The maximum number of simulated_annealing  iterations. Increase this
+        value if the objective function is very complicated with high
+        dimensions.
+    local_search_options : dict, optional
+        Extra keyword arguments to be passed to the local minimizer
+            ``scipy.optimize.minimize()`` Some important options could be:
+            method : str
+                The minimization method (e.g. ``"L-BFGS-B"``)
+            args : tuple
+                Extra arguments passed to the objective function (``func``) and
+                its derivatives (Jacobian, Hessian).
+    initial_temp : float, optional
+        The initial temperature, use higher values to facilitates a wider
+        search of the energy landscape, allowing simulated_annealing  to escape
```

Comment:
double space between ``simulated_annealing`` and ``to escape``.

Reply:
Include a phrase describing the default too.

Reply:
Thanks. Done.